test/hotspot/jtreg/serviceability/jvmti/vthread/PopFrameTest/libPopFrameTest.cpp

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "jvmti.h"
+#include "jvmti_common.h"
+
+extern "C" {
+
+static jvmtiEnv *jvmti = NULL;
+static jmethodID mid_B = NULL;
+static jrawMonitorID monitor = NULL;
+static volatile bool bp_sync_reached = false;
+
+static void JNICALL
+Breakpoint(jvmtiEnv *jvmti, JNIEnv *jni, jthread thread,
+           jmethodID method, jlocation location) {
+  jvmtiError err;
+
+  if (method != mid_B) {
+    fatal(jni, "Breakpoint: Failed with wrong location: expected in method TestTask.B()");
+  }
+  err = jvmti->ClearBreakpoint(mid_B, 0);
+  check_jvmti_status(jni, err, "Breakpoint: Failed in JVMTI ClearBreakpoint");
+
+  LOG("Breakpoint: In method TestTask.B(): before sync section enter\n");
+
+  err = jvmti->RawMonitorEnter(monitor);
+  check_jvmti_status(jni, err, "Breakpoint: Failed in RawMonitorEnter");
+
+  bp_sync_reached = true;
+
+  // wait for notify from notifyAtBreakpoint
+  err = jvmti->RawMonitorWait(monitor, 0);
+  check_jvmti_status(jni, err, "Breakpoint: Failed in RawMonitorWait");
+
+  err = jvmti->RawMonitorExit(monitor);
+  check_jvmti_status(jni, err, "Breakpoint: Failed in RawMonitorExit");
+
+  err = jvmti->PopFrame(thread);
+  LOG("Main: popFrame: PopFrame returned code: %s (%d)\n", TranslateError(err), err);
```

Comment:
check_jvmti_status prints return code and translated error if fails. So this line is not needed,

Reply:
This log is not for error handling but for logging, so I'd like to keep it.
The error code can be not printed but I also prefer to keep it to have it in sync with error handling.